### PR TITLE
Basic implementation of language server

### PR DESCRIPTION
**Pull request by aljazerzen**
This PR adds a language server can be started by `python -m edb.language_server.main` which will then speak LSP over stdin and stdout streams.

It will:
- parse and report any syntactic errors in the opened edgedb files,
- find `dbschema/` directory in the workspace,
- parse and compile the schema,
- compile and edgedb files against that schema,
- if any DDL is found, it will apply that DDL during each file compilation.

Example:

```
# dbschema/default.esdl

module default {
    type Player {
        property username: str;
    };
};
```

```
# a.edgeql

create TYPE Clan {
    create property name -> str;
};

alter type Player {
    create required LINK clan: Clan;
};

select default::Player {
    username,
    clan: {
        name
    }
}
filter .username ilike '%pickle%' and .age > 10;
```

image

TODO:
- [ ] when a schema is changed, recompile it

### CODE REVIEW

**Comment by aljazerzen** [comment]
To check it out, one must:
- checkout this PR into your dev env,
- checkout [language-server branch of edgedb-editor-plugin](https://github.com/edgedb/edgedb-editor-plugin/tree/language-server),
- edgedb-editor-plugin in vscode, using the same python venv that your dev env is installed in,
- launch "Run Extension" task,
- now a new vscode window should open and you can navigate to an edgedb project.

File: pyproject.toml [hunk 1]
```diff
@@ -41,6 +41,9 @@ test = [
     # Needed for testing asyncutil
     'async_solipsism==0.5.0',
 
+    # Needed for language server
+    'pygls~=1.3.1',
+
```
Comment (aljazerzen) [inline pyproject.toml:46]: I'm not sure where to specify this. It should not be packaged into edgedb-server and it is not really needed for tests.

Comment (elprans) [reply pyproject.toml:46]: Put it into another feature, not `test`: 

```toml
lsp = [
    pygls~=1.3.1
]
```

**Comment by msullivan** [comment]
What should we do to test this? Some tests that talk to the language server directly?

**Comment by aljazerzen** [comment]
Probably yes. Run `python -m edb.language_server.main` in a subprocess and feed it LSP requests.

**Review by msullivan — APPROVED** [review]

**Pull request merged**